### PR TITLE
virttest:Optimize code for windows guest balloon test

### PR DESCRIPTION
**Pull request by aliang123**
virttest.utils_misc:add a function "get_used_mem"

virttest.utils_test.qemu: add a function "get_guest_used_mem"

Signed-off-by: Aihua Liang <aliang@redhat.com>

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2641,6 +2641,23 @@ def get_free_mem(session, os_type):
     return int(free)
 
 
+def get_used_mem(session, os_type):
```
Comment (CongLi) [inline virttest/utils_misc.py]: Only windows?

File: virttest/utils_test/qemu.py [hunk 2]
```diff
@@ -1622,6 +1622,19 @@ def get_guest_free_mem(cls, vm):
...
+        """
+        Guest OS reported used memory size in MB.
+
+        :param vm: VM Object
+        :return: used memory report by guest OS in MB
+        """
+        os_type = vm.params.get("os_type")
+        timeout = float(vm.params.get("login_timeout", 600))
+        session = vm.wait_for_login(timeout=timeout)
+        return utils_misc.get_used_mem(session, os_type)
```
Comment (CongLi) [inline virttest/utils_test/qemu.py]: Need close session in this func.

File: virttest/utils_misc.py (hunk 1 above)
Comment (aliang123) [inline virttest/utils_misc.py]: thanks for your comment, i'll update it.

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -2641,6 +2641,27 @@ def get_free_mem(session, os_type):
...
+    :return string: usedspace M-bytes
+    """
+    if os_type == "windows":
+        cmd = "systeminfo | findstr /C:'Virtual Memory: In Use'"
+        output = session.cmd_output(cmd)
+        used = "%sM" % re.findall("\d+", output)[0]
+    else:
+        cmd = "free -m | grep 'Mem'"
+        output = session.cmd_output(cmd)
+        used = "%sM" % output.split()[2]
```
Comment (CongLi) [inline virttest/utils_misc.py]: Need judge output and used is not None.
if output:
    used = ****
     if used:
           float()

**Comment by CongLi** [comment]
LGTM,
acked-by: Cong Li <coli@redhat.com>

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -2641,6 +2641,32 @@ def get_free_mem(session, os_type):
...
+            output = session.cmd_output(cmd)
+            match = re.search(r'Virtual\sMemory:\sIn\sUse:\s+(\d,?\d+)\sMB', output)
+            if match:
+                used = "%sM" % (''.join(match.group(1).split(",")))
+        else:
+            cmd = "free -m | grep 'Mem'"
+            output = session.cmd_output(cmd)
+            used = "%sM" % output.split()[2]
+        used = float(normalize_data_size(used, order_magnitude="M"))
+        return int(used)
```
Comment (suqinhuang) [inline virttest/utils_misc.py]: it's there any difference between int(float(a)) and int(a)

Comment (aliang123) [inline virttest/utils_misc.py]: ok, i'll update it.

Comment (xutian) [inline virttest/utils_misc.py]: bad idea to try so many code.

File: virttest/utils_test/qemu/__init__.py [hunk 5]
```diff
@@ -462,6 +462,22 @@ def get_guest_free_mem(cls, vm):
         session = vm.wait_for_login(timeout=timeout)
```
Comment (ngu-niny) [inline virttest/utils_test/qemu/__init__.py]: @aliang123  The dir seems wrong, can't be applied:

[root@hp-dl385g7-04 avocado-vt]# pwd
/home/staf-kvm-devel/workspace/avocado-vt
[root@hp-dl385g7-04 avocado-vt]# git am 753.patch
Applying: virttest:Add get_used_mem for windows guest balloon test
error: virttest/utils_test/qemu/__init__.py: does not exist in index

Comment (aliang123) [inline virttest/utils_test/qemu/__init__.py]: please update your code to the latest, apply the patch and test.

File: virttest/utils_misc.py [hunk 6]
```diff
@@ -2641,6 +2641,29 @@ def get_free_mem(session, os_type):
...
+
+    :parm session: shell Object.
+    :parm os_type: os type (eg. linux or windows)
+
+    :return string: usedspace M-bytes
+    """
+    if os_type == "windows":
+        cmd = "systeminfo"
+        pattern = r'Virtual\sMemory:\sIn\sUse:\s+(\d,?\d+)\sMB'
+
```
Comment (xutian) [inline virttest/utils_misc.py]: useless blank space

File: virttest/utils_misc.py [hunk 7]
```diff
@@ -2641,6 +2641,28 @@ def get_free_mem(session, os_type):
...
+    Get Used memory for given OS.
+
+    :parm session: shell Object.
+    :parm os_type: os type (eg. linux or windows)
+
+    :return string: usedspace M-bytes
+    """
+    if os_type == "windows":
+        cmd = "systeminfo"
+        pattern = r'Virtual\sMemory:\sIn\sUse:\s+(\d,?\d+)\sMB'
```
Comment (xutian) [inline virttest/utils_misc.py]: it better to replace it to r'Virtual\sMemory: In Use:(.+)'

**Comment by xutian** [comment]
looks ok to me, merged thanks @aliang123

**Pull request merged**